test/jsx.js

```diff
@@ -30,6 +30,61 @@ describe('jsx', () => {
 		`);
 	});
 
+	it('should collapse whitespace', () => {
+		expect(renderJsx(
+			<p>a<a>b</a></p>
+		)).to.equal(dedent`
+			<p>
+				a
+				<a>b</a>
+			</p>
+		`);
+
+		expect(renderJsx(
+			<p>
+				a{' '}
+				<a>b</a>
+			</p>
+		)).to.equal(dedent`
+			<p>
+				a
```

Comment:
I'm not sure if the expected result is correct. It's missing the trailing space from the JSX version.